packages/firebase_admob/android/src/main/java/io/flutter/plugins/firebaseadmob/FirebaseAdMobPlugin.java

```diff
@@ -24,6 +27,12 @@
 
   RewardedVideoAdWrapper rewardedWrapper;
 
+  static Function<UnifiedNativeAd, View> nativeAdGenerator;
```

Comment:
Please, avoid static usage for anything with a View or context. It may become a memory leak quite easily.

Reply:
I'm not sure what you mean by this. The user should only be passing a `View` through a static method and setting it to a member `ViewGroup`. How would the `View` leak?

Reply:
That anonymous function in the Activity will keep an implicit reference to that Activity. That anonymous function is used as factory in the plugin and it is kept as a static reference in the plugin class. It will cause the Activity to leak. A better way to get a reference is to avoid the static reference and make the Activity implement an interface. All of this without using `static` anywhere.

Reply:
I disagree that an anonymous function will keep a reference of the object instantiating it. However, in this context, the function does reference the `Activity` that it is in, so it would keep a reference. This may have been what you meant.

I also agree that your approach with the interface would be cleaner especially as we move away from `static` variables with the v2 embedding migration. I set up a `NativeAdFactory` interface instead.